langchain4j-pgvector/src/main/java/dev/langchain4j/store/embedding/pgvector/JSONMetadataHandler.java

```diff
@@ -0,0 +1,92 @@
+package dev.langchain4j.store.embedding.pgvector;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import dev.langchain4j.data.document.Metadata;
+import dev.langchain4j.store.embedding.filter.Filter;
+
+import java.lang.reflect.Type;
+import java.sql.*;
+import java.util.*;
+
+import static dev.langchain4j.internal.ValidationUtils.ensureNotNull;
+
+/**
+ * This class handle JSON and JSONB Filter mapping
+ */
+public class JSONMetadataHandler implements MetadataHandler {
+    private static final Gson GSON = new Gson();
```

Comment:
Please switch to using `dev.langchain4j.internal.JsonCodec` instead of GSON directly. This is an abstraction over different JSON implementations and we need it to be able to avoid dragging the dependency on GSON into Quarkus.